http.go

```diff
@@ -42,16 +34,14 @@ func (conn SplunkConnection) httpPost(url string, data *url.Values) (string, err
 }
 
 func (conn SplunkConnection) httpCall(url string, method string, data *url.Values) (*http.Response, error) {
-        client := httpClient()
-
         var payload io.Reader
         if data != nil {
           payload = bytes.NewBufferString(data.Encode())
         }
 
         request, err := http.NewRequest(method, url, payload)
         conn.addAuthHeader(request)
-        response, err := client.Do(request)
+        response, err := conn.HttpClient.Do(request)
```

Comment:
just add here a check if conn.HttpClient == nil , use Default Client, so it won't break backward